File: src/main/java/de/kenseiclan/mc/smpchatplugin/config/ConfigHelperImpl.java
package de.kenseiclan.mc.smpchatplugin.config;

import lombok.Getter;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.plugin.Plugin;

import java.util.ArrayList;
import java.util.List;

public class ConfigHelperImpl implements ConfigHelper {
    private final Plugin plugin;

    @Getter
    private List<ChatGroup> chatGroups;

    public ConfigHelperImpl(final Plugin plugin) {
        this.plugin = plugin;
        this.chatGroups = loadGroups();
    }

    private List<ChatGroup> loadGroups() {
        final List<ChatGroup> groups = new ArrayList<>();
        ConfigurationSection groupSection = plugin.getConfig().getConfigurationSection("chat.groups");
        if (groupSection != null) {
            for (String key : groupSection.getKeys(false)) {
                String prefix = groupSection.getString(key + ".prefix");
                groups.add(new ChatGroup(key, prefix));
            }
        }
        return groups;
    }

    @Override
    public void reloadGroups() {
        this.chatGroups = loadGroups();
    }
}

File: src/main/java/de/kenseiclan/mc/smpchatplugin/config/ConfigHelperImpl.kt
package de.kenseiclan.mc.smpchatplugin.config;

import org.bukkit.plugin.Plugin

class ConfigHelperImpl(private val plugin: Plugin) : ConfigHelper {
    var chatGroups: List<ChatGroup> = loadGroups()

    private fun loadGroups(): List<ChatGroup> {
        val groups = mutableListOf<ChatGroup>()
        val groupSection = plugin.config.getConfigurationSection("chat.groups")
        groupSection?.getKeys(false)?.forEach { key ->
            val prefix = groupSection.getString("$key.prefix")
            groups.add(ChatGroup(key, prefix ?: ""))
        }
        return groups
    }

    override fun reloadGroups() {
        chatGroups = loadGroups();
    }
}